satpy/readers/seviri_l2_grib.py

```diff
@@ -153,7 +153,7 @@ def get_dataset(self, dataset_id, dataset_info):
 
                 # Retrieve values and metadata from the GRIB message, masking the values equal to missing_value
                 xarr = self._get_xarray_from_msg(gid)
-                xarr.where(xarr.data == missing_value, np.NaN)
+                xarr.data = xr.where(xarr.data == missing_value, np.nan, xarr.data)
```

Comment:
This seems a little off to me. Since you are accessing `xarr.data` then it would be clearer to me to use dask's `da.where`. As far as I can tell though the original line should have been `xarr = xarr.where(xarr == missing_value, np.nan)`. Is there a reason that won't work here?

Reply:
hi, thanks for the comments. In the original code it didnt set the missing value data points to nan and i couldnt figure out why so i used a similar line of code i had in another reader and this worked ok. 
i can update the reader to try the line of code you mention

Reply:
do you have an example of how you would use the available_datasets method - not sure how we would use this

Reply:
I'm not saying you should do it in this PR necessarily unless you want to explore it as a simpler version of the YAML code. There are a couple readers who use it currently, but the best documentation is currently in the base class itself:

https://github.com/pytroll/satpy/blob/93f0233f9e4675d516de70dd36183dc6320fa82e/satpy/readers/file_handlers.py#L163-L248

The only time this doesn't work is if you want to define additional metadata in the YAML that isn't available in the file (or can't be easily determined in code).

Reply:
hi, the amendment to the original line of code, masking out the missing values doesnt work. The returned array still includes missing values thus my data values are incorrect. 
I updated the new line of code to use dask array where and this works as expected.
any thoughts

Reply:
My original suggestion was wrong, it should have been `xarr != missing_value`. What you have now is fine, but a little "indirect" since there is a method that is more commonly used in Satpy readers and is supposed to work just fine. If what you have now works then leave it (it is at least consistent with its use of dask and xarray objects).